addons/website_sale_delivery/static/src/js/website_sale_delivery.js

```diff
@@ -51,8 +51,9 @@ publicWidget.registry.websiteSaleDelivery = publicWidget.Widget.extend({
             // Update discount of the order
             $('#order_discounted').find('.oe_currency_value').text(result.new_amount_order_discounted);
 
-            // We are in freeshipping, so every carrier is Free
-            $('#delivery_carrier .badge').text(_t('Free'));
+            // We are in freeshipping, so every carrier is Free but we don't
+            // want to replace error message by 'Free'
+            $('#delivery_carrier .badge:not(.carrier-error)').text(_t('Free'));
```

Comment:
Should not this commit be in stable versions?

Reply:
yes I forgot to mention but 3 first commits can go in 12.0 if you are ok with it. (`remove useless check`, `don't replace error message by 'Free'`, `show 'Free' instead of '$0'`)
(Not < 12.0 as there was some refactoring on 12.0, it would be annoying to forward port)

Reply:
Only "don't replace error message by 'Free'" seems to be a bug to me ;)

Reply:
Yes, I was viewing your commit on the wrong commit.
Can you confirm you are ok for both 3 first commits (mentionned above) to go in 12.0?
Then I will open a PR in 12.0 with those and the current PR will have to wait for the forward port.

Reply:
I will keep this only in master as there was a refactoring of this file between 12.0 and master + it is indeed a 'bug' but having an error on a carrier is not supposed to happens as it is a bad configuration on carrier.
I replaced `carrier-error` by `o_wsale_delivery_carrier_error` tho.